js/ui/calendar/ui.calendar.selection.strategy.js

```diff
@@ -0,0 +1,62 @@
+class CalendarSelectionStrategy {
+    constructor(component) {
+        this.calendar = component;
+    }
+
+    dateOption(optionName) {
```

Comment:
I see that some methods are not overrided in other strategies, so y do we need to implement it?

Reply:
I added some methods like this one in base strategy just to make code in strategies a bit more clear, so we can call `this.dateOption() instead of calling `this.calendar._dateOption()`.